### DISCUSSION TITLE
can rich be used on MicroPython or pycopy

### DISCUSSION THREAD

**Discussion by varioustoxins** [Textualize/rich#2065] [Q&A]
I wanted to write a small (yet colourful) command line installer. So far I have been working with bash and some 3rd part libraries. Last nigh it struck me rich and a small foot print python (micropython/pycopy) might be a good fit. Is this possible?

**Comment by willmcgugan** [comment]
I haven’t tried, but I can’t see why it wouldn’t work!

**Reply by varioustoxins** [reply]
Thanks, that really good to hear. I think I will have a go. I think the only problem maybe that micropython is at 3.4 with 'selected later features'. So I may need a 3to3 library! is that called 9... or maybe 12.24?

**Reply by varioustoxins** [reply]
so had a go and got quite a long way but ran into the problem that rich uses enums and enums use metaclasses and ... metaclasses are not supported, bummer

**Reply by varioustoxins** [reply]
maybe ways around but will require a fake IntEnum class...

**Reply by varioustoxins** [reply]
or possibly the python27 back port...

**Reply by varioustoxins** [reply]
nope metaclasses are the killer, no metaclasses no enums...

**Reply by varioustoxins** [reply]
next alternative build a bistro with standard python but much/some of the standard library stripped away...
